count/include/count_min.hpp

```diff
@@ -0,0 +1,189 @@
+#ifndef COUNT_MIN_HPP_
+#define COUNT_MIN_HPP_
+
+#include <iterator>
+#include "common_defs.hpp"
+
+namespace datasketches {
+
+  /*
+   * C++ implementation of the CountMin sketch data structure of Cormode and Muthukrishnan.
+   * [1] - http://dimacs.rutgers.edu/~graham/pubs/papers/cm-full.pdf
+   * The template type W is the type of the vector that contains the weights of the objects inserted into the sketch,
+   * not the type of the input items themselves.
+   * @author Charlie Dickens
+   */
+
+template<typename W>
+class count_min_sketch{
+  static_assert(std::is_arithmetic<W>::value, "Arithmetic type expected");
+public:
+
+  /**
+  * Creates an instance of the sketch given parameters num_hashes, num_buckets and hash seed, `seed`.
+  * @param num_hashes : number of hash functions in the sketch. Equivalently the number of rows in the array
+  * @param num_buckets : number of buckets that hash functions map into. Equivalently the number of columns in the array
+  * @param seed for hash function
+  *
+  * The items inserted into the sketch can be arbitrary type, so long as they are hashable via murmurhash.
+  * Only update and estimate methods are added for uint64_t and string types.
+  */
+  count_min_sketch(uint16_t num_hashes, uint32_t num_buckets, uint64_t seed = DEFAULT_SEED) ;
```

Comment:
The number of hashes does not need to be a 16-bit int.
The largest 5 bit int permits a failure probability of about  1/e^31 = 3E-12 which is plenty small enough.
the smallest datatype we can use is uint8_t so use that instead.